lib/concurrent/thread_safe/util/array_hash_rbx.rb

```diff
@@ -15,6 +15,12 @@ def self.allocate
             obj.send(:_mon_initialize)
             obj
           end
```

Comment:
Is changing `allocate` still needed then?
Or should `_mon_initialize` simply be called in `initialize`?
cc @pitr-ch